syntax = "proto3";

import 'nanopb.proto';

// Defines the format for data or settings reports
message Report {
    oneof report {
        NodeSettingsReport nodeSettingsReport = 1;
    }
}

// Informs about the current node settings
message NodeSettingsReport {
    string firmware_version = 1 [(nanopb).max_length = 32];
    
    string node_name = 2 [(nanopb).max_length = 32];
    string node_contact = 3 [(nanopb).max_length = 32];
    
    float location_lat = 4;
    float location_long = 5;

    bool wifi_password_set = 6;

    uint32 wake_interval = 7;

    bytes app_eui = 8 [(nanopb).max_size = 8, (nanopb).fixed_length = true];
    bytes dev_eui = 9 [(nanopb).max_size = 8, (nanopb).fixed_length = true];

    bool app_key_set = 10;

    bool link_check_enabled = 11;
    bool adr_enabled = 12;
    uint32 spreading_factor = 13;
    sint32 tx_power = 14;
}

// LorawanStatusReport, SensorsReport